### ISSUE TITLE
Caching response status separatly? 

### ISSUE THREAD

**Issue by bewatts** [oesmith/puffing-billy#223]
Hello!

We've got a sticky problem - We're hitting against an internal API that returns the result of a long-running process.  While the process is running, it returns a 202.  When it resolves, it returns a 200. 

We're running into an issue were Billy caches the initial 202 request, and we're stuck with that cached result.  

Any guidance?  Is resolving this a new feature request?  

We'd prefer to not manually specify the response from the API, and let Billy cache the results as they actually come in.

**Comment by bewatts** [comment]
Hmmmmm...that would be hard.  This is basically the only API we're caching. 

Would a feature quests of not caching certain response codes be reasonable?   

Right now there's a `non_successful_cache_disabled` flag.   There could be a `c.status_codes_to_avoid_caching = [202]` flag?  

This feels like a general problem: Any other API that uses 202s to indicate that the response is not ready yet would have this problem.

**Comment by ronwsmith** [comment]
Hmm, the gem wasn't really intended for polling APIs.  Does it work if we remove `202` from the `successful_status?` helper?

**Comment by bewatts** [comment]
Yes, that would solve our problem - that should prevent the response from being cached.

**Comment by ronwsmith** [comment]
Since 202 means still processing, I'm fine with changing that if you want to submit a PR.
